mask/eolearn/mask/cloud_mask.py

```diff
@@ -292,3 +297,592 @@ def get_s2_pixel_cloud_detector(threshold=0.4, average_over=4, dilation_size=2,
+                          the required 10. Default value:  `True`.
+        :type all_bands: bool
+        :param processing_resolution: Resolution to be used during the computation of cloud probabilities and masks.
+                                      Resolution is given as a pair of x and y resolutions. If a single value is
+                                      given, it is used for both dimensions.
+                                      Default is `None` (source resolution).
+        :type processing_resolution: (str, str) or (int, int) or str or int
+        :param max_proc_frames: Maximum number of frames (including the target, for multi-temporal classification)
+                                considered in a single batch iteration (To keep memory usage at agreeable levels,
+                                the task operates on smaller batches of time frames). Default value:  `11`.
+        :type max_proc_frames: int
+        :param mono_features: Tuple of keys to be used for storing cloud probabilities and masks of the mono classifier.
+                              The probabilities are added to the `eopatch.data` attribute dictionary, while masks are
+                              added to `eopatch.mask`. By default, none of them are added.
+        :type mono_features: (str | None, str | None)
+        :param multi_features: Tuple of keys used for storing cloud probabilities and masks of the multi classifier.
+                               The probabilities are added to the `eopatch.data` attribute dictionary, while masks are
+                               added to `eopatch.mask`. By default, none of them are added.
+        :type multi_features: (str | None, str | None)
+        :param mask_feature: Name of the output intersection feature. The masks are added to the `eopatch.mask`
+                             attribute dictionary. Default value: `'CLM_INTERSSIM'`. If None, the intersection
+                             feature is not computed.
+        :type mask_feature: str | None
+        :param mono_threshold: Cloud probability threshold for the mono classifier. Default value: `0.4`.
+        :type mono_threshold: float
+        :param multi_threshold: Cloud probability threshold for the multi classifier. Default value: `0.5`.
+        :type multi_threshold: float
+        :param average_over: Size of the pixel neighbourhood used in the averaging post-processing step.
+                             A value of `0` or `None` skips this post-processing step. Default value: `1`.
+        :type average_over: int or None
+        :param dilation_size: Size of the dilation post-processing step. A value of `0` or `None` skips
+                              this post-processing step. Default value: `1`.
+        :type dilation_size: int or None
+        """
+
+        # Load classifiers
+        if mono_classifier is None or multi_classifier is None:
```

Comment:
This should always be true, or not? If not, there should rather be a check for this anyway.

Reply:
It's not always true. Both classifiers may be provided by the user.

What do you mean by there should be a check?

Reply:
The outside if can be removed without problems if that's what you meant. I agree that there's no need for `classifier_dir` to only be initialized when needed.

The code might be more readable this way.